### PR TITLE
Authentication Documentaton

### LINKED ISSUE #174: Document the files in mifopay/auth package
**Summary:**
Document the files in mifospay/auth package: https://github.com/openMF/mobile-wallet/tree/master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth

### PR DESCRIPTION
**Pull request by PraneethVankayala**
Fixes #174 

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Apply the `AndroidStyle.xml` style template to your code in Android Studio.

- [x] Run the unit tests with `./gradlew check` to make sure you didn't break anything

- [x] If you have multiple commits please combine them into one commit by squashing them.

### CODE REVIEW

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/presenter/LoginPresenter.java [hunk 1]
```diff
@@ -18,9 +18,12 @@
 import javax.inject.Inject;
 
 /**
- * Created by naman on 16/6/17.
+ * This class is the Presenter component of the auth package.
+ * @author  naman
+ * @since  16-June-17.
  */
 
+
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/presenter/LoginPresenter.java]:
```suggestion
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/ui/LoginActivity.java [hunk 2]
```diff
@@ -36,9 +36,12 @@
 import butterknife.OnClick;
 
 /**
- * Created by naman on 16/6/17.
+ * This is the UI component of the auth package
+ * @author  naman
+ * @since  16-June-17.
  */
 
+
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/ui/LoginActivity.java]:
```suggestion
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/ui/LoginActivity.java [hunk 3]
```diff
@@ -83,19 +89,29 @@ public void onLoginClicked() {
...
     @Override
     public void loginSuccess() {
         hideProgressDialog();
         Utils.hideSoftKeyboard(this);
         startPassCodeActivity();
     }
 
+    /**
+     * An overriden function implemented if login is failed.
+     *@param message A variable of type String
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/ui/LoginActivity.java]:
```suggestion
     * @param message A variable of type String
```

**Comment by PraneethVankayala** [comment]
@luckyman20 updated

**Comment by PraneethVankayala** [comment]
@luckyman20 sorry for making such mistakes since I solved the issue which I am encountering since yesterday I made some typo's in a hurry to send the pull request

**Comment by PraneethVankayala** [comment]
@luckyman20 sorry for not checking the grammatical mistakes, I think now I had updated all the mistakes I did please review this once

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/presenter/LoginPresenter.java [hunk 4]
```diff
@@ -38,14 +40,20 @@ public LoginPresenter(UseCaseHandler useCaseHandler, PreferencesHelper preferenc
...
+     */
     @Override
     public void attachView(BaseView baseView) {
         mLoginView = (AuthContract.LoginView) baseView;
         mLoginView.setPresenter(this);
     }
 
-
+    /**
+     * A function to  authenticate a user based on the credentials
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/presenter/LoginPresenter.java]:
```suggestion
     * A function to authenticate a user based on the credentials
```

File: mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/presenter/LoginPresenter.java [hunk 5]
```diff
@@ -38,14 +40,20 @@ public LoginPresenter(UseCaseHandler useCaseHandler, PreferencesHelper preferenc
...
+     * @param username A variable of the type String
```
Comment (luckyman20) [inline mifospay/src/main/java/org/mifos/mobilewallet/mifospay/auth/presenter/LoginPresenter.java]:
```suggestion
     * @param username Username given by the user.
```
Do same for the password.

**Comment by PraneethVankayala** [comment]
@luckyman20 please review this

**Comment by PraneethVankayala** [comment]
changes updated

**Comment by luckyman20** [comment]
@PraneethVankayala Please squash your commits.

**Comment by PraneethVankayala** [comment]
@luckyman20 updated

**Comment by PraneethVankayala** [comment]
@luckyman20 squashed the commits

**Comment by PraneethVankayala** [comment]
@luckyman20 please review this

**Comment by luckyman20** [comment]
@PraneethVankayala Please squash your commits.

**Comment by jsahil730** [comment]
@PraneethVankayala are you still working on this? If yes, please first resolve all the conflicts so that we may proceed any further, else please close this PR.

**Comment by luckyman20** [comment]
Closing this because of inactivity.

**Pull request closed without merging**